### DISCUSSION TITLE
Ingesting OPERA data via pcp_combine

### DISCUSSION THREAD

**Discussion by lindsayrblank** [dtcenter/METplus#1590] [File I/O] [answered]
Hello all,

I am attempting to ingest EUMETNET OPERA netCDF data using the pcp_combine tool. I am running the MET v10.0 Docker image on an HPC using Singularity. Is there a way to do this?

Here is what I have tried so far. 
OPERA data is netCDF 4/HDF 5. When I run the following command: `pcp_combine -add T_PASH21_C_EUOC_20220424000000.nc test_opera.nc -field 'product="COMP"; quantity="ACRR";'` I get the following error: ` get_field() -> can't open data file "T_PASH21_C_EUOC_20220424000000.nc"`

I also tried running plot_data_plane. When I run the following command: `plot_data_plane T_PASH21_C_EUOC_20220424000000.nc test_opera.ps 'name="ACRR"; level="(*,*)";'` I get the following error: `plot_data_plane -> file "T_PASH21_C_EUOC_20220424000000.nc" not a valid data file`. 

OPERA files don't have a clear `name` and `level` setting so I was messing with those in the field string. 

Thanks,
Lindsay

**Comment by jprestop** [comment, marked as answer]
Hi @lindsayrblank!

You may already be familiar with the "file_type" option, but in case you aren't, please take a look at the information about the “file_type” entry in [this section](https://met.readthedocs.io/en/develop/Users_Guide/config_options.html?highlight=file_type#fcst) of the Configuration Options.  As you may already know MET handles only a few flavors of NetCDF:

- NetCDF file created by another MET tool
- NetCDF created by running the p_interp or wrf_interp utility on WRF output.
- NetCDF files following the Climate Forecast (CF) convention.

As far as I know, MET has not been modified to read in the EUMETNET OPERA NetCDF data.  If the NetCDF data is not in one of the above formats, Python embedding may be necessary to read in this data.  Please take a look to see if the NetCDF data follows the Climate Forecast (CF) convention and then try adding "file_type = NETCDF_NCCF;".  If you'd like us to take a look, please feel free to [post your data to the ftp server](https://github.com/dtcenter/METplus/discussions/954) and then follow up here to let us know you've posted it.

**Reply by lindsayrblank** [reply]
Hi Julie, 

Thanks for the information! It appears that OPERA data are not in CF compliant. I'll take a look at python embedding in order to make OPERA data CF compliant. 

Thanks!

**Reply by jprestop** [reply]
Thanks @lindsayrblank.  If you feel my response answered your question, could you please select "Mark as answer" and then I'll lock this conversation to prevent future posts, to encourage users to create new discussions for new questions.   If you would like to leave this open for now in case you have follow up questions about this data, that is ok too.
